### PR TITLE
[Android Auto] Migrate CarRouteLine to MapboxNavigationApp

### PR DESCRIPTION
**Pull request by kmadsen**
### Description

Fixing `CarRouteLine` to use `MapboxNavigationObserver`. Am also removing `MainCarContext` from the `CarLocationRenderer` constructor as it is no longer needed.

### CODE REVIEW

File: libnavui-androidauto/src/main/java/com/mapbox/androidauto/car/preview/CarRouteLine.kt [hunk 1]
```diff
@@ -77,27 +81,30 @@ class CarRouteLine(
...
                     .build()
             )
 
+            val locationPlugin = mapboxCarMapSurface.mapSurface.location
             locationPlugin.addOnIndicatorPositionChangedListener(onPositionChangedListener)
-            mainCarContext.mapboxNavigation.registerRouteProgressObserver(routeProgressObserver)
+            MapboxNavigationApp.registerObserver(navigationObserver)
             coroutineScope.launch {
                 carRoutesProvider.navigationRoutes.collect { onRoutesChanged(it) }
             }
```
Comment (Zayankovsky) [inline libnavui-androidauto/src/main/java/com/mapbox/androidauto/car/preview/CarRouteLine.kt:110]: Not related to this PR, but there seems to be an error, we launch a new job for every style load. We should also cancel the previous one.

File: libnavui-androidauto/src/main/java/com/mapbox/androidauto/car/preview/CarRouteLine.kt [hunk 2]
```diff
@@ -129,10 +148,9 @@ class CarRouteLine(
 
     private fun onRoutesChanged(routes: List<NavigationRoute>) {
         logAndroidAuto("CarRouteLine onRoutesChanged ${routes.size}")
-        mainCarContext.mapboxCarMap.carMapSurface?.getStyle()?.let { style ->
-            if (routes.isNotEmpty()) {
```
Comment (Zayankovsky) [inline libnavui-androidauto/src/main/java/com/mapbox/androidauto/car/preview/CarRouteLine.kt:133]: We don't perform this check anymore, is it expected?

File: libnavui-androidauto/src/main/java/com/mapbox/androidauto/car/preview/CarRouteLine.kt [hunk 3]
```diff
@@ -129,10 +148,9 @@ class CarRouteLine(
...
-                val routesMetadata =
-                    mainCarContext.mapboxNavigation.getAlternativeMetadataFor(routes)
+        carMapSurface?.getStyle()?.let { style ->
```
Comment (Zayankovsky) [inline libnavui-androidauto/src/main/java/com/mapbox/androidauto/car/preview/CarRouteLine.kt:151]: We can pass the style from `onAttached`.

**Review by Zayankovsky — APPROVED** [review]

File: libnavui-androidauto/src/main/java/com/mapbox/androidauto/car/preview/CarRouteLine.kt (hunk 2 above)
Comment (kmadsen) [reply libnavui-androidauto/src/main/java/com/mapbox/androidauto/car/preview/CarRouteLine.kt:133]: my logic here is wrong, am updating

File: libnavui-androidauto/src/main/java/com/mapbox/androidauto/car/preview/CarRouteLine.kt (hunk 1 above)
Comment (kmadsen) [reply libnavui-androidauto/src/main/java/com/mapbox/androidauto/car/preview/CarRouteLine.kt:110]: yeah this is a mess

Comment (kmadsen) [reply libnavui-androidauto/src/main/java/com/mapbox/androidauto/car/preview/CarRouteLine.kt]: ok fixed it. nice catch!

Every time the style changed there is an extra call to `onRoutesChanged`, so if you change the style 6 times each re-route would trigger `onRoutesChanged` 6 times

Comment (kmadsen) [reply libnavui-androidauto/src/main/java/com/mapbox/androidauto/car/preview/CarRouteLine.kt]: it's fixed

File: libnavui-androidauto/src/main/java/com/mapbox/androidauto/car/preview/CarRouteLine.kt (hunk 3 above)
Comment (kmadsen) [reply libnavui-androidauto/src/main/java/com/mapbox/androidauto/car/preview/CarRouteLine.kt]: Needed to take this out of the scope that repeated when the style changed.

Technically we could do something like this
``` kotlin
.combine(styleFlow, this@CarRouteLine::onRoutesChanged).collect()

private fun onRoutesChanged(routes: List<NavigationRoute>, style: Style?): List<NavigationRoute> {
    style?.let { style ->
```

But I didn't like the "transform" function where the routes are returned, and it needs to do a null check anyways

Going with this.
``` kotlin
private fun onRoutesChanged(routes: List<NavigationRoute>) {
    styleFlow.value?.let { style ->
```

**Pull request merged**